src/NServiceBus.Gateway/obsoletes-v4.cs

```diff
@@ -0,0 +1,21 @@
+﻿#pragma warning disable 1591
+
+namespace NServiceBus.Gateway
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+
+    [ObsoleteEx(
+        TreatAsErrorFromVersion = "8.0.0",
+        RemoveInVersion = "9.0.0",
+        ReplacementTypeOrMember = nameof(DataReceivedOnChannelEventArgs))]
+    public class DataReceivedOnChannelArgs : EventArgs
+    {
+        public IDictionary<string, string> Headers { get; set; }
+
+        public Stream Data { get; set; }
+    }
```

Comment:
The version numbers are wrong, and I think we can remove the implementation?

Also, how is it that we don't need to suppress IDE1006 here?

```suggestion
    [ObsoleteEx(
        TreatAsErrorFromVersion = "4.0.0",
        RemoveInVersion = "5.0.0",
        ReplacementTypeOrMember = nameof(DataReceivedOnChannelEventArgs))]
    public class DataReceivedOnChannelArgs { }
```

Reply:
> Also, how is it that we don't need to suppress IDE1006 here?

I think it's because of https://github.com/Particular/NServiceBus.Gateway/pull/293#discussion_r688537251 - I think those lines are removing this file from compilation, which is definitely not what we want.

Reply:
duh, fixed now!

Reply:
and it make the compile error show up like you said